src/main/java/com/it4045/pokemonviewer/enterprise/service/IPokemonService.java

```diff
@@ -9,5 +9,6 @@
 public interface IPokemonService {
 
     Pokemon fetchByName(String pokemonName);
+    Pokemon fetchByNumber(int pokemonNumber);
     List<Pokemon> fetchPokemons() throws IOException;
 }
```

Comment:
First technical suggestion: Added functionality to retrieve a Pokemon by number in the Pokemon service class.

I noticed that the program was using the API to retrieve Pokemon by their name, so I decided to add to this functionality by enabling it to retrieve Pokemon by their number as well. Name and number are both ways of identifying a Pokemon, but to my understanding, retrieving its number might be helpful for a number of reasons - in particular, because Pokemon which are evolutions of each other tend to have adjacent numbers, so this functionality may be helpful for retrieving groups of Pokemon which are related in this way.